### ISSUE TITLE
Warnings, sometimes errors, when running sort() on DNAStringSet

### ISSUE THREAD

**Issue by LTLA** [Bioconductor/Biostrings#7]
Running the following bit of innocuous code:

```r
library(Biostrings)
a <- DNAStringSet(c("AAA", "AA", "A"))
sort(a)
```

... yields the correct result, but an odd warning:

```
  A DNAStringSet instance of length 3
    width seq
[1]     1 A
[2]     2 AA
[3]     3 AAA
Warning message:
In order(..., na.last = na.last, decreasing = decreasing) :
  "order" method for XRawList objects ignores the 'na.last' argument
```

More concerning is the fact that the code fails if _flowCore_ is loaded prior to _Biostrings_:

```r
library(flowCore)
library(Biostrings)
a <- DNAStringSet(c("AAA", "AA", "A"))
sort(a)
```

... which produces the error:

```
Error in validObject(.Object) : 
  invalid class “MethodWithNext” object: Error : evaluation nested too deeply: infinite recursion / options(expressions=)?
```

... presumably because there is a `sort` in _flowCore_ that overrides something.

```
R Under development (unstable) (2017-10-27 r73632)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

Matrix products: default
BLAS: /home/cri.camres.org/lun01/Software/R/trunk/lib/libRblas.so
LAPACK: /home/cri.camres.org/lun01/Software/R/trunk/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
[1] Biostrings_2.47.9   XVector_0.19.8      IRanges_2.13.28    
[4] S4Vectors_0.17.35   BiocGenerics_0.25.3 flowCore_1.45.7    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15       lattice_0.20-35    matrixStats_0.53.1 mvtnorm_1.0-7     
 [5] corpcor_1.6.9      rrcov_1.4-3        grid_3.5.0         pcaPP_1.9-73      
 [9] graph_1.57.1       zlibbioc_1.25.0    robustbase_0.92-8  Biobase_2.39.2    
[13] DEoptimR_1.0-8     compiler_3.5.0     cluster_2.0.6     
```

**Comment by hpages** [comment]
Hi @LTLA ,

I fixed the warning in XVector 0.19.9:
  https://github.com/Bioconductor/XVector/commit/d36d0212a6ff3a7dbd189f52c15f51c6d6a4083b

For flowCore breaking dispatch on the `sort()` generic defined in BiocGenerics, this could be avoided if they were defining a `"sort"` method for the same generic (i.e. for `BiocGenerics::sort`) instead of defining their own `sort()` generic. Do you want to let them know about this?

Thanks,
H.

**Comment by LTLA** [comment]
Thanks @hpages, done.
